controlplane/src/core/repositories/OrganizationInvitationRepository.ts

```diff
@@ -0,0 +1,150 @@
+import { and, asc, eq } from 'drizzle-orm';
+import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import * as schema from '../../db/schema.js';
+import { organizationInvitations, organizations, users } from '../../db/schema.js';
+import { OrganizationDTO, OrganizationInvitationDTO, UserDTO } from '../../types/index.js';
+import { OrganizationRepository } from './OrganizationRepository.js';
+import { UserRepository } from './UserRepository.js';
+
+export class OrganizationInvitationRepository {
+  constructor(private db: PostgresJsDatabase<typeof schema>) {}
+
+  // returns the members who have pending invites to the provided organization.
+  public async getPendingInvitationsOfOrganization(input: {
+    organizationId: string;
+  }): Promise<OrganizationInvitationDTO[]> {
+    const pendingInvites = await this.db
+      .select({
+        userID: users.id,
+        email: users.email,
+      })
+      .from(organizationInvitations)
+      .innerJoin(users, eq(users.id, organizationInvitations.userId))
+      .where(
+        and(
+          eq(organizationInvitations.organizationId, input.organizationId),
+          eq(organizationInvitations.acceptedInvite, false),
+        ),
+      )
+      .orderBy(asc(organizationInvitations.createdAt))
+      .execute();
+
+    return pendingInvites;
+  }
+
+  // returns the organizations to which the user has a pending invite.
+  public async getPendingInvitationsOfUser(input: { userId: string }): Promise<OrganizationDTO[]> {
+    const pendingOrgInvites = await this.db
+      .select({
+        id: organizations.id,
+        name: organizations.name,
+        slug: organizations.slug,
+        creatorUserId: organizations.createdBy,
+        createdAt: organizations.createdAt,
+      })
+      .from(organizationInvitations)
+      .innerJoin(organizations, eq(organizations.id, organizationInvitations.organizationId))
+      .innerJoin(users, eq(users.id, organizationInvitations.userId))
+      .where(and(eq(users.id, input.userId), eq(organizationInvitations.acceptedInvite, false)))
+      .execute();
+
+    const userInvitations = pendingOrgInvites.map((org) => ({
+      id: org.id,
+      name: org.name,
+      slug: org.slug,
+      creatorUserId: org.creatorUserId,
+      createdAt: org.createdAt.toISOString(),
+    }));
+
+    return userInvitations;
+  }
+
+  public async getPendingOrganizationInvitation(input: {
```

Comment:
Why just a single one?

Reply:
To check if the invitation already exists

Reply:
thx